scenedetect/scene_manager.py

```diff
@@ -491,13 +491,19 @@ def get_event_list(self, base_timecode):
                 for start, end in self._event_list]
 
 
-    def _process_frame(self, frame_num, frame_im):
+    def _process_frame(self, frame_num, frame_im, callback=None):
         # type(int, numpy.ndarray) -> None
         """ Adds any cuts detected with the current frame to the cutting list. """
         for detector in self._detector_list:
-            self._cutting_list += detector.process_frame(frame_num, frame_im)
+            cuts = detector.process_frame(frame_num, frame_im)
+            if len(cuts) and callback:
+                callback(frame_im)
```

Comment:
Should the frame number or timecode be passed to the callback as well?

Reply:
I can do the frame number, but where do I obtain the timecode?

Reply:
@Breakthrough are you available?

Reply:
Good point, I forgot the scene manager only deals with frame numbers.  Sorry for the delay in getting back to you on this one, will definitely be trying to get this in for the next release though!

Reply:
No problem at all. I pushed a commit with `frame_num`.